packages/strapi-generate-api/json/routes.json.js

```diff
@@ -97,46 +97,20 @@ function generateCollectionTypeRoutes({ route, name }) {
  */
 
 module.exports = scope => {
-  const routes =
-    scope.contentTypeKind === 'singleType'
-      ? generateSingleTypeRoutes({
-          route: scope.route,
-          name: scope.name,
-        })
-      : generateCollectionTypeRoutes({
-          route: scope.route,
-          name: scope.name,
-        });
+  let routes = [];
+  if (!scope.args.plugin) {
```

Comment:
Hello @alexandrebodin & @petersg83, What's the reason for disabling the API routes generation inside plugins? I've deleted that `if (!scope.args.plugin) {` locally and it seems to generate/merge existing routes ok.

Reply:
As the plugin models don't get auto generated controllers those default routes point to nothing. There was no reason to generate them :)

Reply:
In that case, it also brokes the functionality of the `strapi generate:api order --plugin shop` command,  which should create a new API inside the plugin, it doesn't generate the routes for it automatically.

Reply:
There should be added an extra condition on that `if`, that verifies if the `generate:model` was used.